awx/api/views/__init__.py

```diff
@@ -317,6 +317,54 @@ def get(self, request, format=None):
         return Response(dashboard_data)
 
 
+class MetricsView(APIView):
+
+    view_name = _("Metrics")
+    swagger_topic = 'Metrics'
+    
+    renderer_classes = [renderers.BrowsableAPIRenderer,
+                        renderers.PlainTextRenderer,
+                        JSONRenderer]
+    
+    def get(self, request, format=None):
+        ''' Show Metrics Details '''
+        
+        # Temporary Imports
+        from awx.main.models.organization import UserSessionMembership
+        from django.contrib.sessions.models import Session
+
+        def _prepare_data(data):
+            metrics = ''
+            for metric in data:
+                metrics += metric + '\n'
+            return metrics
+        
+        # Add active/expired, or only query active sessions
+        
+        total_sessions = Session.objects.all().count()
+        active_sessions = Session.objects.filter(expire_date__gte=now()).count()
+        
+        
+        api_sessions = UserSessionMembership.objects.all().count()
+        channels_sessions = total_sessions - api_sessions
+        expired_sessions = total_sessions - active_sessions
+        
+        data = []
+        data.append("# HELP awx_sessions_active counter A count of active sessions.")
+        data.append("# TYPE awx_sessions_active counter")
+        data.append("sessions.active_sessions {0} ".format(total_sessions))
+        data.append("# TYPE awx_sessions_websocket counter")
+        data.append("sessions.websocket_sessions {0} ".format(channels_sessions))
+        data.append("# TYPE awx_sessions_api counter")
+        data.append("sessions.api_sessions {0} ".format(api_sessions))
+        data.append("# TYPE sessions.active_sessions counter")
+        data.append("sessions.active_sessions {0}".format(active_sessions))
+        data.append("# TYPE sessions.expired_sessions counter")
+        data.append("sessions.expired_sessions {0}".format(expired_sessions))
+        
+        return Response(_prepare_data(data))
```

Comment:
You could probably replace the `prepare_data` function with just:

```python
"\n".join(metrics)
```

Reply:
nice, that's a good point.  I'll be taking this whole section out eventually, this is just test data from when I was trying to hook up Prometheus to test things out.  I'll be writing a separate function to prepare the data from the analytics stuff, but I'll split that off into another PR